microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/core/jaxrs/JaxRsContext.java

```diff
@@ -75,19 +95,48 @@ public String getApplicationPath() {
 	}
 
 	/**
-	 * Set the quarkus.http.root-path property
+	 * Set the @ApplicationPath annotation valye
 	 *
 	 * @param applicationPath as the @ApplicationPath annotation value
 	 */
 	public void setApplicationPath(String applicationPath) {
 		this.applicationPath = applicationPath;
 	}
 
+	/**
+	 * Get the Java CodeLens Context
+	 *
+	 * @return the Java CodeLens Context
+	 */
+	public JavaCodeLensContext getJavaCodeLensContext() {
+		return javaCodeLensContext;
+	}
+
+	/**
+	 * Set the quarkus.http.root-path property
+	 *
+	 * @param javaCodeLensContext
+	 */
+	public void setJavaCodeLensContext(JavaCodeLensContext javaCodeLensContext) {
+		this.javaCodeLensContext = javaCodeLensContext;
+	}
+
+	/**
+	 * @param the progress monitor
+	 * @throws CoreException
+	 */
+	public void retrieveApplicationPath(IProgressMonitor monitor) throws CoreException {

```

Comment:
The main idea is to have a `getApplicationPath` which search the application if it is not done otherwise it returns the cached result and remove the `setApplicationPath`

In other words, replace `retrieveApplicationPath` with:

```java
public String getApplicationPath() {
   if (applicationPath != null) {
     return applicationPath;
  }
  applicationPath = // use the search engine to retrieve the value.
}
```

Reply:
The `getApplicationPath() ` will be called when base Url is computed.

Reply:
If the user updates the `@ApplicationPath` value while the quarkus project is running, this will prevent the CodeLens from updating with the quarkus endpoint live update. Do we want to do that?

Reply:
Update, turns out the application path updates on the codelens, however, a new jaxrscontext is made when beginCodeLens is run, so the search engine is always run regardless, which circles back to our original issue.

Reply:
A new JaxRsContext insance is created when a new display code lens is done and it's the expected behavior.